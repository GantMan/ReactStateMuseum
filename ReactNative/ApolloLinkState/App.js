import React, { Component } from 'react'
import { StyleSheet, Text, ScrollView } from 'react-native'
import { ApolloProvider, Query } from 'react-apollo';
import { client } from './client';
import { ALL_ITEMS } from './query';
import ListItems from './Components/listItems'
import AddItems from './Components/addItems'

export default class App extends Component {
  state = {
    newItemName: null
  }

  setNewItemName = (value) => {
    this.setState({newItemName: value})
  }

  addItem = (store, items) => {
    const newItem = this.state.newItemName;
    this.setState({ newItemName: '' }, () =>
      store.writeData({
        data: { allItems: items.concat(newItem) }
    }))
  }

  clear = (store) => {
    store.writeData({ data: { allItems: [] }})
  }

  render () {
    return (
      <ApolloProvider client={client}>
        <Query query={ALL_ITEMS}>
        {
          ({data, client: store}) => (
            <ScrollView
              contentContainerStyle={styles.container}
              keyboardShouldPersistTaps='always'
            >
              <Text style={styles.welcome}>
                Welcome to apollo-link-state
              </Text>
              <AddItems
                addItem={() => this.addItem(store, data.allItems)}
                setNewItemName={this.setNewItemName}
                newItemName={this.state.newItemName}
                clear={() => this.clear(store)}
              />
              <ListItems allItems={data.allItems} />
            </ScrollView>
          )
        }
        </Query>
      </ApolloProvider>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})
