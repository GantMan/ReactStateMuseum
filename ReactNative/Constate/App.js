import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  Text,
  StatusBar,
} from 'react-native';

import AddItem from './Components/addItem';
import ListItems from './Components/listItems';
import { PackingProvider } from './Constate/packingContext';

const App = () => {
  return (
    <PackingProvider>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          keyboardShouldPersistTaps="handled"
          style={styles.scrollView}
        >
          <Text style={styles.welcome}>Welcome to Constate!</Text>
          <AddItem />
          <ListItems />
        </ScrollView>
      </SafeAreaView>
    </PackingProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20,
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

export default App;
