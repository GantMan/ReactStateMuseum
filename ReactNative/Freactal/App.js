import React from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'

import ListItems from './Components/listItems'
import AddItems from './Components/addItems'
import { withState } from './State/listWrap'

export const App = () => {
  return (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps='always'
    >
      <Text style={styles.welcome}>
        Welcome to Freactal
      </Text>
      <AddItems />
      <ListItems />
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})

export default withState(App)
