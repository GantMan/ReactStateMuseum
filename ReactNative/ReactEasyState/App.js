import React from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'
import { view } from 'react-easy-state'
import { items } from './store'
import ListItems from './Components/listItems'
import AddItems from './Components/addItems'

const App = view(() => {
  const { newItem, allItems, setNewItemValue, addItem, clearItems } = items

  return (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps='always'
    >
      <Text style={styles.welcome}>
        Welcome to React Easy State
      </Text>
      <AddItems
        addItem={addItem}
        setNewItemText={setNewItemValue}
        value={newItem}
        clear={clearItems}
      />
      <ListItems allItems={allItems} />
    </ScrollView>
  )
})

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})

export default App
