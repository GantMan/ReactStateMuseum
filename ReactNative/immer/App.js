import React from 'react';
import { StyleSheet, Text, ScrollView } from 'react-native';
import produce from 'immer';
import ListItems from './Components/listItems';
import AddItems from './Components/addItems';

export default class App extends React.Component {
  state = {
    allItems: ['nachos', 'burritos', 'hot dog'],
    newItemName: ''
  };

  addItem = () => {
    this.setState(
      produce(state => {
        state.allItems.push(state.newItemName);
        state.newItemName = '';
      })
    );
  };

  setNewItemName = value => {
    this.setState(
      produce(state => {
        state.newItemName = value;
      })
    );
  };

  clear = () => {
    this.setState(
      produce(state => {
        state.allItems = [];
      })
    );
  };

  render() {
    return (
      <ScrollView contentContainerStyle={styles.container} keyboardShouldPersistTaps="always">
        <Text style={styles.welcome}>Welcome to immer</Text>
        <AddItems
          addItem={this.addItem}
          setNewItemName={this.setNewItemName}
          newItemName={this.state.newItemName}
          clear={this.clear}
        />
        <ListItems allItems={this.state.allItems} />
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: "#F5FCFF",
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  }
});
