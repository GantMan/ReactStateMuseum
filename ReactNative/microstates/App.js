import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  ScrollView
} from 'react-native'

import ListItems from './Components/listItems'
import AddItems from './Components/addItems'
import State from '@microstates/react';

// Define your state model
class ListModel {
  allItems = [String];
  newItemText = String;
}

export default class App extends Component {
  render() {
    return (
      <State
        type={ListModel}
        value={{ allItems: ["nachos", "burritos", "hot dog"] }}
      >
        {list => (
          <ScrollView
            contentContainerStyle={styles.container}
            keyboardShouldPersistTaps="always"
          >
            <Text style={styles.welcome}>Welcome to microstates</Text>
            <AddItems
              addItem={() =>
                list.allItems.push(list.state.newItemText).newItemText.set('')
              }
              setNewItemName={(value) => list.newItemText.set(value)}
              newItemName={list.state.newItemText}
              clear={() => list.allItems.set([])}
            />
            <ListItems allItems={list.state.allItems} />
          </ScrollView>
        )}
      </State>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})
