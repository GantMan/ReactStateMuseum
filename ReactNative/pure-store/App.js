import React, { Component } from 'react'
import { StyleSheet, Text, ScrollView } from 'react-native'
import { SimpleList, AddPackingItem } from 'packlist-components/native'
import createStore from "pure-store"


const store = createStore({
  allItems: ["nachos", "burritos", "hot dog"],
  newItemName: ""
})

export default class App extends Component {
  constructor () {
    super()
    store.subscribe(()=> this.forceUpdate())
  }

  addItem = ()=> {
    store.update(state=> {
      state.allItems.push(state.newItemName)
      state.newItemName = ""
    })
  }

  setName = n => store.update({ newItemName: n })
  clear   = ()=> store.update({ allItems: [] })

  render = ()=> (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps='always'
    >
      <Text style={styles.welcome}>
        Welcome to pure-store
      </Text>
      <AddPackingItem
        value={store.state.newItemName}
        addItem={this.addItem}
        setNewItemText={this.setName}
        clear={this.clear}
      />
      <SimpleList
        value={store.state.allItems}
      />
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  }
})
