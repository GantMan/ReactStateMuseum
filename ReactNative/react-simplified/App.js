import React from "react";
import { Component } from "react-simplified";
import { StyleSheet, Text, ScrollView } from "react-native";

import ListItems from "./Components/listItems";
import AddItems from "./Components/addItems";

export default class App extends Component {
  allItems = ["nachos", "burritos", "hot dog"];
  newItemName = "";

  addItem() {
    this.allItems.push(this.newItemName);
    this.newItemName = "";
  }

  setNewItemName(value) {
    this.newItemName = value;
  }

  clear() {
    this.allItems = [];
  }

  render() {
    return (
      <ScrollView contentContainerStyle={styles.container} keyboardShouldPersistTaps="always">
        <Text style={styles.welcome}>Welcome to setState</Text>
        <AddItems
          addItem={this.addItem}
          setNewItemName={this.setNewItemName}
          newItemName={this.newItemName}
          clear={this.clear}
        />
        <ListItems allItems={this.allItems} />
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: "#F5FCFF",
    padding: 20
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  }
});
