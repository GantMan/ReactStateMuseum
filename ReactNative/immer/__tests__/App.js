import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../App';

test('renders correctly', () => {
  const wrapper = shallow(
    <App />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});