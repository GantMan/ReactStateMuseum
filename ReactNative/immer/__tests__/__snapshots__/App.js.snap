// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ScrollViewMock
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "padding": 20,
    }
  }
  keyboardShouldPersistTaps="always"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to immer
  </Text>
  <AddItems
    addItem={[Function]}
    clear={[Function]}
    newItemName=""
    setNewItemName={[Function]}
  />
  <AddItems
    allItems={
      Array [
        "nachos",
        "burritos",
        "hot dog",
      ]
    }
  />
</ScrollViewMock>
`;
